### DISCUSSION TITLE
关于CP Editor内置特殊评测器的 bug

### DISCUSSION THREAD

**Discussion by wscqwq** [cpeditor/cpeditor-cn#33] [提问]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

使用 `CP Editor` 时，发现自定义比较器报错。
经过检查，是由于 `testlib.h` 版本过久导致的。
评测器编译时的报错信息如下：
```
[08:53:49] [评测器] [开始编译评测器]
[08:53:51] [评测器] [
编译评测器时发生错误：
In file included from C:/Users/20144/AppData/Local/Temp/CP Editor-haopDG/checker.cpp:1:
C:/Users/20144/AppData/Local/Temp/CP Editor-haopDG/testlib.h: In member function 'std::vector<T> random_t::distinct(int, T, T)':
C:/Users/20144/AppData/Local/Temp/CP Editor-haopDG/testlib.h:1136:9: error: 'uint64_t' was not declared in this scope
 1136 |         uint64_t n = to - from + 1;
      |         ^~~~~~~~
C:/Users/20144/AppData/Local/Temp/CP Editor-haopDG/testlib.h:193:1: note: 'uint64_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
  192 | #       include <windows.h>
  +++ |+#include <cstdint>
  193 | #   else
C:/Users/20144/AppData/Local/Temp/CP Editor-haopDG/testlib.h:1137:30: error: 'n' was not declared in this scope
 1137 |         if (uint64_t(size) > n)
      |                              ^
C:/Users/20144/AppData/Local/Temp/CP Editor-haopDG/testlib.h:1142:32: error: 'n' was not declared in this scope
 1142 |             expected += double(n) / double(n - i + 1);
      |                                ^
C:/Users/20144/AppData/Local/Temp/CP Editor-haopDG/testlib.h:1144:31: error: 'n' was not declared in this scope
 1144 |         if (expected < double(n)) {
      |                               ^
]
```
我没有找到 `CP Editor` 使用的是哪个 `testlib.h`，应该是版本过旧导致的。（新版的testlib引入了 `#include<cstdint>`，保证 `uint64_t` 可用）。
我不知道如何修改 `CP Editor` 使用的 `testlib.h`。
![](https://github.com/wscqwq/image/blob/main/%E5%B1%8F%E5%B9%95%E6%88%AA%E5%9B%BE%202023-10-15%20085809.png)

### Expected Behavior

编译评测器不会出错

### Steps To Reproduce

1. 打开软件
2. 修改评测器一栏
3. 报错

### CP Editor Version

6.11.1&6.10.3

### OS

Windows 10

### Are you willing to contribute to this?

- [ ] I'm ready to fix this (and will open a PR soon)
- [ ] I'd like to have a try (with the help of the maintainers)
- [X] No, thanks

### Anything else?

我在这里下载：https://cpeditor.org/zh/download/ 和 https://github.com/cpeditor/cpeditor/releases

**Comment by wscqwq** [comment]
图片
![image-20231015091854383](https://img2023.cnblogs.com/blog/3107168/202310/3107168-20231015091855070-566824950.png)

**Comment by ouuan** [comment]
和 #28 重复
